src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFun.kt

```diff
@@ -47,155 +40,179 @@ abstract class ComickFun(
+        )
+    }
 
     /** Latest Manga **/
     override fun latestUpdatesRequest(page: Int): Request {
-        return searchMangaRequest(
-            page = page,
-            query = "",
-            filters = FilterList(
-                SortFilter("", getSortsList, defaultLatestSort),
-            ),
-        )
+        val url = "$apiUrl/v1.0/search?sort=uploaded&limit=$limit&page=$page&tachiyomi=true"
+        return GET(url, headers)
     }
 
-    override fun latestUpdatesParse(response: Response) = searchMangaParse(response)
+    override fun latestUpdatesParse(response: Response) = popularMangaParse(response)
 
     /** Manga Search **/
     override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
-        if (!query.startsWith(SLUG_SEARCH_PREFIX)) {
-            return super.fetchSearchManga(page, query, filters)
+        return if (query.startsWith(SLUG_SEARCH_PREFIX)) {
+            // url deep link
+            val slugOrHid = query.substringAfter(SLUG_SEARCH_PREFIX)
+            val manga = SManga.create().apply { this.url = "/comic/$slugOrHid#" }
+            fetchMangaDetails(manga).map {
+                MangasPage(listOf(it), false)
+            }
+        } else if (query.isEmpty()) {
+            // regular filtering without text search
+            client.newCall(searchMangaRequest(page, query, filters))
+                .asObservableSuccess()
+                .map(::searchMangaParse)
+        } else {
+            // text search, no pagination in api
+            if (page == 1) {
+                client.newCall(querySearchRequest(query))
+                    .asObservableSuccess()
+                    .map(::querySearchParse)
+            } else {
+                Observable.just(paginatedSearchPage(page))
+            }
         }
+    }
+
+    private fun querySearchRequest(query: String): Request {
+        val url = "$apiUrl/v1.0/search?limit=300&page=1&tachiyomi=true"
+            .toHttpUrl().newBuilder()
+            .addQueryParameter("q", query.trim())
+            .build()
+
+        return GET(url, headers)
+    }
 
-        val slugOrHid = query.substringAfter(SLUG_SEARCH_PREFIX)
-        val manga = SManga.create().apply { this.url = "/comic/$slugOrHid#" }
-        return fetchMangaDetails(manga).map {
-            MangasPage(listOf(it), false)
+    private fun querySearchParse(response: Response): MangasPage {
+        searchResponse = response.parseAs()
+
+        return paginatedSearchPage(1)
+    }
+
+    private fun paginatedSearchPage(page: Int): MangasPage {
+        val mangas = mutableListOf<SManga>()
+        val endRange = ((page * limit) - 1).let { if (it <= searchResponse.lastIndex) it else searchResponse.lastIndex }
+
+        for (i in (((page - 1) * limit)..endRange)) {
+            mangas.add(
+                searchResponse[i].toSManga(),
+            )
         }
+        return MangasPage(mangas, endRange < searchResponse.lastIndex)
```

Comment:
```suggestion
        val end = min(page * limit, searchResponse.size)
        val entries = searchResponse.subList((page - 1) * limit, end)
            .map(SearchManga::toSManga)
        return MangasPage(entries, end < searchResponse.size)
```

You'll also need to `import kotlin.math.min`.